case_studies/CARD/Bug-Drug Code.R

```diff
@@ -17,61 +17,120 @@ untar("broadstreet-v3.3.0.tar", exdir = "CARD_data")
 
 
 # Map CARD Short Name
-# Install and Load dplyr
-if (!require("dplyr")) {
-  install.packages("dplyr")
-  library(dplyr)
-} else {
-  library(dplyr)
+# Install and Load dplyr and readr
+packages <- c("dplyr", "readr")
+
+for (pkg in packages) {
+  if (!require(pkg, character.only = TRUE)) {
+    install.packages(pkg)
+    library(pkg, character.only = TRUE)
+  } else {
+    library(pkg, character.only = TRUE)
+  }
+}
+
+# Parse the required files using readr::read_delim
+aro_index <- read_delim("CARD_data/aro_index.tsv", delim = "\t", col_names = TRUE)
+antibiotics_data <- read_delim("CARD_data/shortname_antibiotics.tsv", delim = "\t", col_names = TRUE)
+pathogens_data <- read_delim("CARD_data/shortname_pathogens.tsv", delim = "\t", col_names = TRUE)
+
+
+# Extract pathogen, gene, drug, and include Protein.Accession from 'CARD Short Name'
+library(dplyr)
+library(purrr)
+library(stringr)
+
+# Extract pathogen, gene, drug, and include Protein.Accession from 'CARD Short Name'
+extract_card_info <- function(card_short_name, drug_class, `Protein Accession`, `DNA Accession`) {
+  # Split the CARD Short Name by underscores
+  split_names <- unlist(strsplit(card_short_name, "_"))
+  
+  # Initialize variables with defaults
+  pathogen <- NA
+  gene <- NA
+  drug <- drug_class  # Default to Drug Class column
+  
+  # Determine the information based on the split names and patterns
```

Comment:
can you share an example file (snippet pre and post name cleanup)?

Reply:
> can you share an example file (snippet pre and post name cleanup)?

Hello @jananiravi , by this do you mean I should  use the View() function in R to allow for the visual inspection of the dataset before and after processing

Reply:
No, I meant snapshots or example data stored locally (as part of the commit) to be able to run the code and check locally.